llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp

```diff
@@ -409,7 +409,7 @@ bool RISCVCallLowering::lowerReturnVal(MachineIRBuilder &MIRBuilder,
   splitToValueTypes(OrigRetInfo, SplitRetInfos, DL, CC);
 
   RVVArgDispatcher Dispatcher{&MF, getTLI<RISCVTargetLowering>(),
-                              F.getReturnType()};
+                              ArrayRef(F.getReturnType())};
```

Comment:
Do we need to construct an ArrayRef explicitly?

Reply:
Yeah we do, since `ArrayRef` is also a template class, it can't deduct the type, you can try a simple example:
```
    1 template <typename T>
    2 class abc {
    3 public:
    4   abc(T);
    5 };
    6
    7 template <typename T>
    8 void test(abc<T> a);
    9
   10 int main()
   11 {
   12   test(1);
   13 }
```
the code can't compile since `void test(abc<T> a)` doesn't know that 1 can be implicitly constructed, it only know that 1's type is not equal to `abc` lol~

Reply:
In this example you could explicitly set T in the call (`test<int>(1)`) but… you can't do that in a constructor call like in your case of `RVVArgDispatcher`, so I'm afraid the explicit cast will have to remain.